### DISCUSSION TITLE
Markdown support?

### DISCUSSION THREAD

**Discussion by maxn0d3x** [codex-team/editor.js#1870] [Ideas]
this editor.js supports Markdown editing or even converting option?

**Comment by gohabereg** [comment]
Hi @NiNJAD3vel0per !

We will work on some converting options in the next releases

**Reply by jshmllr** [reply]
I find it hard to believe a tool in '23 /'24 doesn't support .md — for example: https://ui.toast.com/tui-editor works out of the box -- even CK Editor https://github.com/ckeditor/ckeditor5/blob/d9d69bacc6717321bf044ff4d047eb0f8e2e4a33/packages/ckeditor5-markdown-gfm/src/markdown.ts#L18 has a package for .md

When we want to be as close to the source of content as possible, this seems like a barebones necessity of a text product.

**Reply by nam-co** [reply]
I just wish it could load and save simple html

**Comment by maxn0d3x** [comment]
> Hi @NiNJAD3vel0per !
> 
> We will work on some converting options in the next releases

Can't Wait!

**Comment by koliyo** [comment]
Related to https://github.com/codex-team/editor.js/issues/682

**Comment by iiDestiny** [comment]
want to support markdown

**Comment by cschweda** [comment]
Markdown, please. :)

**Comment by cong5** [comment]
Would you like support markdown?

**Comment by appernetic** [comment]
Yes, markdown, please! :-)

**Comment by alxlion** [comment]
Is someone developing this feature ?

**Comment by timtody** [comment]
any updates on this?

**Comment by Zulcom** [comment]
I tied to make my implementation with [mdast](https://github.com/syntax-tree/mdast) but this is really hard to maintain "CLEAN data that easy to sanitize, extend and integrate with your logic" with markdown.
Example of clean data:
>  On **this page** you can see it in action — try to edit this text.

will be 

```
{
  "type": "parahraph",
  "data": {
    "text": " On <b>this page</b> you can see it in action — try to edit this text."
  }
}
```
Which means you literally should re-implement ALL built-in inline tools to support markdown because we have HTML tags inside "CLEAN data" at this point. And idk how to do it without forking and changing core code

**Comment by GauravRajSharma** [comment]
Actively watching this. Markdown input -> blocks -> output Markdown would be a godsent feature for editor.js

**Comment by tvvignesh** [comment]
> I tied to make my implementation with [mdast](https://github.com/syntax-tree/mdast) but this is really hard to maintain "CLEAN data that easy to sanitize, extend and integrate with your logic" with markdown.
> Example of clean data:
> 
> > On **this page** you can see it in action — try to edit this text.
> 
> will be
> 
> ```
> {
>   "type": "parahraph",
>   "data": {
>     "text": " On <b>this page</b> you can see it in action — try to edit this text."
>   }
> }
> ```
> 
> Which means you literally should re-implement ALL built-in inline tools to support markdown because we have HTML tags inside "CLEAN data" at this point. And idk how to do it without forking and changing core code

@Zulcom I was wondering how they are doing bold and italics and looks like the HTML tags are indeed mixed with the data as seen in the online demo. They say about inline elements in the docs and maybe markdown elements are meant to be inline 🤔 :

<img width="388" alt="Capture" src="https://user-images.githubusercontent.com/1165845/92020217-71894700-ed75-11ea-9e2a-809d64753503.PNG">

I guess they would be following the same approach for markdown as well.

**Comment by subwaymatch** [comment]
This would be a killer feature 🙏🔥🌪!

**Comment by taurenshaman** [comment]
MarkdownBlock: add Markdown Block support:  #1500
---
[editorjs-markdown-parser in awesome-editorjs](https://github.com/stejul/editorjs-markdown-parser) is not suitable for me, so I created another one: [markdown-block](https://github.com/tunaltd/blocks-editor/tree/main/src/components/markdown-block)

demo:  
![image](https://user-images.githubusercontent.com/1308706/104009629-55dff280-51e6-11eb-9090-4801e3d32a09.png)


https://user-images.githubusercontent.com/1308706/104009770-7dcf5600-51e6-11eb-9a4b-61d760aa6623.mp4

[youtube](https://www.youtube.com/watch?v=Bsvm65tP4Bs)

**Comment by rlindskog** [comment]
> I tied to make my implementation with [mdast](https://github.com/syntax-tree/mdast) but this is really hard to maintain "CLEAN data that easy to sanitize, extend and integrate with your logic" with markdown.
> Example of clean data:
> 
> > On **this page** you can see it in action — try to edit this text.
> 
> will be
> 
> ```
> {
>   "type": "parahraph",
>   "data": {
>     "text": " On <b>this page</b> you can see it in action — try to edit this text."
>   }
> }
> ```
> 
> Which means you literally should re-implement ALL built-in inline tools to support markdown because we have HTML tags inside "CLEAN data" at this point. And idk how to do it without forking and changing core code

I'm messing around with using [turndown](https://github.com/domchristie/turndown) to parse the paragraph blocks, so far I'm having some good luck. I'll post my progress here, might have a package up in the near future.

**Comment by MatthewCaseres** [comment]
> I'm messing around with using [turndown](https://github.com/domchristie/turndown) to parse the paragraph blocks, so far I'm having some good luck. I'll post my progress here, might have a package up in the near future.

This would be great. unified.js tools might work for this as well.

**Comment by laurent22** [comment]
I'm surprised a converter still doesn't exist 4 years later since it seems trivial to convert back and forth between the JSON format and MD, or am I missing something? Anyone gave it a try and what issues did you have when doing so?

**Comment by tiagozip** [comment]
Still no update?

**Comment by prykris** [comment]
This is a basic necessity for a project like this. Still interested!

**Comment by neSpecc** [comment]
We are going to add this to our Roadmap. If you are interested in this feature and want to expedite its development, now is the best time to support us with a donation on OpenCollective

https://opencollective.com/editorjs

**Comment by tiagozip** [comment]
Does anyone know an inline markdown paragraph extension? I've been trying
to use tiptap instead of editor.js but its kinda hard to make a good editor
with it without using Next.js

Michael Mc ***@***.***> escreveu (segunda, 23/09/2024 à(s)
15:47):

**Reply by prykris** [reply]
> Does anyone know an inline markdown paragraph extension? I've been trying to use tiptap instead of editor.js but its kinda hard to make a good editor with it without using Next.js Michael Mc ***@***.***> escreveu (segunda, 23/09/2024 à(s) 15:47):
> […](#)
> I just wish it could load and save simple html — Reply to this email directly, view it on GitHub <[#1870 (reply in thread)](https://github.com/codex-team/editor.js/discussions/1870#discussioncomment-10727514)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AQ3M5XTJUL6QJ433OKSWTJ3ZYASWPAVCNFSM6AAAAABASENUXSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZSG42TCNA> . You are receiving this because you commented.Message ID: ***@***.***>

I implemented ProseMirror & TipTap editor yesterday. It was smooth sailing. Their project is very mature and extendable. Albeit it could be tedious to setup. As you have to build the UI yourself. Which did not bother me that much, for me it was quite beneficial anyway. It was no brainer to pick this since it can work with no annoying JS Framework which I very much despise.

**Reply by sindras** [reply]
@prykris that sounds awesome! Do you mind sharing your project for reference?

**Comment by lukajose** [comment]
Any updates on this?

**Reply by prykris** [reply]
> Any updates on this?

No. Look for alternatives

**Reply by lukajose** [reply]
What did you do?

**Reply by prykris** [reply]
> What did you do?

I ended up implementing ProseMirror

**Reply by jkazimierczak** [reply]
Milkdown is also [based on prosemirror](https://milkdown.dev/docs/guide/prosemirror-api). You might take a look at it.

**Reply by lukajose** [reply]
wow milkdown looks awesome , why is this not more popular. Thanks for the suggestion!

**Reply by nam-co** [reply]
Milkdown does look good, fast, but one thing I notice, if you add a H6 how do you change it in the UI for a h3? I couldn't find a way, strange

**Reply by sindras** [reply]
> Milkdown is also [based on prosemirror](https://milkdown.dev/docs/guide/prosemirror-api). You might take a look at it.

I would not recommend Milkdown. Notice that it has not received active development since September and there is only one active maintainer (the creator themselves). Take a look at any open issues, none of them receiving response.

**Reply by jkazimierczak** [reply]
I agree that looking at the overall "health" of given project before adopting it such as you mentioned amount of contributors, issues open, how fast (if at all) are these resolved are good metrics to vet the dependency. I do it myself, and this approach is rational IMO.

However, I always found the attitude _"it wasn't updated in a month, it's dead, we must reinvent the wheel/find a new one"_ / _"they don't push on save, the project must be dead"_ a bit funny. Everyone is just a person - if one's decides to take a break, or naturally shift to other project then that's completely fine. Many take a lot of OSS dependencies for granted, giving nothing to little in return (especially those who profit from them). Maintaining OSS was never easy, and will never be. What starts as a passion project often becomes a burden over time, when no one but the author contributes to ever rising demands. Also, from what I see, the main author is mainly focused on BlockSuite and Affine (so other projects).

If you have a luxury of waiting, that's good - observe the health of existing solutions. If you're pressured by any constraint (work/time/contract) pick what's available, evaluate how well it's supported and go with it.

Also, since Milkdown is based on prosemirror, you can see how they solved/implemented features you might want to implement yourself with the prosemirror anyway. You don't have to use it directly, you could as well use it just as a reference.

---

TL;DR: You can use Milkdown as a reference, you don't have to use it directly. Don't follow the hype at all cost, evaluate the suitability and health of the dependency yourself before including it.

**Reply by sindras** [reply]
No objection there. I was stating what's currently happening and the potential risk. In fact, I contributed to the repo by fixing and reporting issues. I also gave a chance for the owner to reply about the future roadmap so more people can contribute. 

https://github.com/Milkdown/milkdown/issues/1552

> Also, from what I see, the main author is mainly focused on BlockSuite and Affine (so other projects).

Isn't this another source of concern? 😄 If _the only maintainer_ is working on something else, what will happen to the old project?

I can't influence everyone's opinion for sure, but based on my own experience and lack of communication, I have decided to move on to a more suitable editor.

**Reply by jkazimierczak** [reply]
Got it. That's a good observation worth sharing and I don't undermine that. I just wanted to share my outlook on it, as I know some folks just follow trends and then they're hit with a surprise. I personally have still quite a lot of time before I tie myself to an editing library, so I'm still on a lookout.

> Isn't this another source of concern? 😄

It definitely is! While these projects seem kind-of related to Milkdown, it isn't said that their advancements will translate onto the Milkdown itself.

> I have decided to move on to a more suitable editor.

Out of curiosity, what did you end up with?

**Reply by sindras** [reply]
All good! You brought up valid points.

I evaluated BlockSuite and Affine as well btw, but as they mentioned on [Overview](https://blocksuite.io/guide/overview.html) - it is still in very early development. BlockSuite is a combination of Excalidraw and Notion, which is too ambitious for my projects.

I also read through [Yoopta editor](https://yoopta.dev/). The editor comes with batteries included. However, the docs choice of words is somewhat off-putting. It also uses slate.js which is still in early stage.

Last weekend, I tried [BlockNote](https://www.blocknotejs.org/). Their docs are more mature, straight to the point, and very transparent about their architecture. They have paid-support option, which shows sustainability and careful planning. It is built on top of TipTap and ProseMirror. It has MPL 2.0 license. I recently [raised an issue](https://github.com/TypeCellOS/BlockNote/issues/1213) on their github repo (same issue with Milkdown [1533](https://github.com/Milkdown/milkdown/issues/1533)), and the author immediately work on it [1216](https://github.com/TypeCellOS/BlockNote/pull/1216).

**Comment by fveiraswww** [comment]
This is a utility that converts clean data (editor js) into HTML (https://github.com/pavittarx/editorjs-html), and it also lets you add custom tools with your own parser (https://github.com/pavittarx/editorjs-html?tab=readme-ov-file#extend-for-custom-blocks). It’s not md, but it’s another way to export your content

**Comment by pranavmappoli** [comment]
Any updates ?

**Comment by MarcosEich** [comment]
It would be really nice to have markdown support in Editor.js
Waiting for good news

**Comment by VityaSchel** [comment]
I ended up converting all my content pages from editor.js to markdown via custom script and using markdown editors/renderers

https://milkdown.dev/ is pretty good one

**Comment by itsthejoker** [comment]
Any updates on markdown support?
